src/Aggregate/AggregateRepository.php

```diff
@@ -142,20 +144,30 @@ public function addAggregateRoot($eventSourcedAggregateRoot)
             );
         }
 
-        $aggregateId = $this->aggregateTranslator->extractAggregateId($eventSourcedAggregateRoot);
+        $domainEvents = $this->aggregateTranslator->extractPendingStreamEvents($eventSourcedAggregateRoot);
 
-        $domainEvents = new ArrayIterator(
-            $this->aggregateTranslator->extractPendingStreamEvents($eventSourcedAggregateRoot)
+        //We make a copy of the aggregate root to be able to extract the aggregate id
+        //without the need to apply the pending domain events to the real aggregate instance
+        //(this is done in the method applyPendingStreamEvents after EventStore.commit)
+        $aggregateRootCopy = $this->aggregateTranslator->reconstituteAggregateFromHistory(
+            $this->aggregateType,
+            new ArrayIterator($domainEvents)
```

Comment:
it should be enough to apply only the first event, no reconstruction of the complete history is needed. Would you like to submit a second PR?